### PR TITLE
[dailymotion] Improve extractor

### LINKED ISSUE #3229: [dailymotion] support embedded link type geo.dailymotion
### Checklist

- [X] I'm reporting a site feature request
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

anywhere

### Example URLs

https://geo.dailymotion.com/player.html?video=x89eyek&mute=true

### Description

This link was originally extracted from https://www.cnews.fr/emission/2022-03-27/en-quete-desprit-du-27032022-1197552

Unsupported URL: https://geo.dailymotion.com/player.html?video=x89eyek&mute=true

### Verbose log

```shell
-v -F "https://geo.dailymotion.com/player.html?video=x89eyek&mute=true"
[debug] Command-line config: ['-v', '-F', 'https://geo.dailymotion.com/player.html?video=x89eyek&mute=true']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.1 [28469edd7] (zip)
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.9.10 (CPython 64bit) - macOS-11.6.5-arm64-arm-64bit
[debug] exe versions: none
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://geo.dailymotion.com/player.html?video=x89eyek&mute=true
[generic] player: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] player: Downloading webpage
[generic] player: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://geo.dailymotion.com/player.html?video=x89eyek&mute=true
Traceback (most recent call last):
  File "/Users/zig/yt-dlp/./yt-dlp/yt_dlp/YoutubeDL.py", line 1403, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/zig/yt-dlp/./yt-dlp/yt_dlp/YoutubeDL.py", line 1473, in __extract_info
    ie_result = ie.extract(url)
  File "/Users/zig/yt-dlp/./yt-dlp/yt_dlp/extractor/common.py", line 639, in extract
    ie_result = self._real_extract(url)
  File "/Users/zig/yt-dlp/./yt-dlp/yt_dlp/extractor/generic.py", line 4017, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://geo.dailymotion.com/player.html?video=x89eyek&mute=true
```

### PR DESCRIPTION
**Pull request by hatienl0i261299**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Close #3229

### CODE REVIEW

File: yt_dlp/extractor/dailymotion.py [hunk 1]
```diff
@@ -94,10 +94,10 @@ class DailymotionIE(DailymotionBaseInfoExtractor):
     _VALID_URL = r'''(?ix)
                     https?://
                         (?:
-                            (?:(?:www|touch)\.)?dailymotion\.[a-z]{2,3}/(?:(?:(?:embed|swf|\#)/)?video|swf)|
+                            (?:(?:www|touch|geo)\.)?dailymotion\.[a-z]{2,3}/(?:(?:(?:(?:embed|swf|\#)/)|player\.html\?)?video|swf)|
                             (?:www\.)?lequipe\.fr/video
                         )
-                        /(?P<id>[^/?_]+)(?:.+?\bplaylist=(?P<playlist_id>x[0-9a-z]+))?
+                        (?:\/|\=)(?P<id>[^/?_&]+)(?:.+?\bplaylist=(?P<playlist_id>x[0-9a-z]+))?
```
Comment (pukkandan) [inline yt_dlp/extractor/dailymotion.py:100]:
```suggestion
                        [/=](?P<id>[^/?_&]+)(?:.+?\bplaylist=(?P<playlist_id>x[0-9a-z]+))?
```

**Review by pukkandan — COMMENTED** [review]
are you sure about this? By trying to capture 2 very different patterns with one regex, there is a much higher possibility of encountering false positives.

File: yt_dlp/extractor/dailymotion.py (hunk 1 above)
Comment (hatienl0i261299) [reply yt_dlp/extractor/dailymotion.py:100]: I thought about it and a tested again with some another case and i think its ok. If you think its not good, i can move geo.dailymotion with different pattern and different class.

Comment (pukkandan) [reply yt_dlp/extractor/dailymotion.py:100]: I assume u r replying to:

> By trying to capture 2 very different patterns with one regex, there is a much higher possibility of encountering false positives.

Just wanted to make you aware of the risk. If you think it's a non-issue, it's fine

But this suggestion is unrelated to that comment

**Comment by dirkf** [comment]
To avoid false positives, you can make the match of the path and query parts depend on the matched domain or subdomain by wrapping the specific domain alternative in `(?P<tag>...)` and then using `(?(tag)match when tag is found|match when not found)` in the trailing part.

**Comment by hatienl0i261299** [comment]
To make sure there are no regex errors, i move it to new class to handle for dailymotion_player

**Comment by hatienl0i261299** [comment]
To make sure there are no regex errors, i will move dailimotion_player to new class.

**Comment by pukkandan** [comment]
Moving `_real_extract` to the base class is a bad idea here

File: yt_dlp/extractor/dailymotion.py [hunk 2]
```diff
@@ -115,6 +115,25 @@ class DailymotionIE(DailymotionBaseInfoExtractor):
...
+            'duration': 2756,
+            'timestamp': 1648383669,
+            'upload_date': '20220327',
+            'uploader': 'CNEWS',
+            'uploader_id': 'x24vth',
+            'age_limit': 0,
+            'view_count': int,
+            'like_count': int,
+            'tags': ['en_quete_d_esprit'],
+            'thumbnail': 'https://s2.dmcdn.net/v/Tncwi1YGKdvFbDuDY/x1080',
```
Comment (dirkf) [inline yt_dlp/extractor/dailymotion.py:135]: This should be made less specific, eg (supposing `Tncwi1YGKdvFbDuDY` is an immutable thumbnail ID for this video):
```
            'thumbnail': r're:https?://\w+\.dmcdn\.net/v/Tncwi1YGKdvFbDuDY/x1080',
```

**Pull request merged**